### PR TITLE
Config consolidation

### PR DESCRIPTION
**Pull request by krazeelazy**
### I have:
- [X] [Searched](https://github.com/BoopChat/boop/pulls) the bugtracker for similar pull requests
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] Covered the code with tests that prove my fix is effective or that my feature works (note that PRs
without tests will be REJECTED)
- [ ] New and existing unit tests pass locally with my changes

## How Has This Been Tested?

I followed the quick start guide in the README.md file. The db was setup and seeded properly. I then ran the app and was able to sign in, view my contact list, view my convo list and send new messages.

---

### What is the purpose of your *pull request*?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Improvement
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of your *pull request* and other information

- Made .sequelizerc file to allow us to use a js file for sequelize config
- No longer using server/config/config in server/app.js
- Added to .env file:
    - DB_NAME
    - DB_USERNAME
    - DB_PASSWORD
    - DB_HOST
    - HOME_URL
    - FACEBOOK_CALLBACK_URL
    - GOOGLE_CALLBACK_URL
    - USER_INFO_EMAIL
    - USER_INFO_FIRST_NAME
    - USER_INFO_LAST_NAME
- No longer using global.gConfig for homeURL, and callbacks for google and facebook. These now come from the .env file
- server/config/config.js now gets the db username, password, name and host from the env vars
- removed sample_config.json since private values now come from .env file so server/config/config.js can be checked into source control
- the seeder now gets its values from the process.env values
- Updated ./README.md and server/README.md file to direct user to set config. vars in .env only
- Updated readme with docker build instructions
- Removed hardcoded db config from docker-compose file in favor of env vars
- Use NODE_ENV to determine env instead of req.app.get("env")
- Sync the name of the env vars used for db config in docker compose files

### CODE REVIEW

File: .sequilezerc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var path = require("path");
+module.exports = { 
+"config": path.resolve("./config", "config.js"),
+"models-path": path.resolve("server/models"),
+"seeders-path": path.resolve("server/seeders"),
+"migrations-path": path.resolve("server/migrations")}
```
Comment (JChris246) [inline .sequilezerc:6]: format? 🤪

File: server/config/config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const dotenv = require("dotenv");
+dotenv.config();
+
+module.exports = {
+    "development": {
```
Comment (davidfowl) [inline server/config/config.js:5]: Do we need the environment here at all?

**Review by JChris246 — CHANGES_REQUESTED** [review]

File: server/config/config.js (hunk 2 above)
Comment (JChris246) [reply server/config/config.js:5]: Probably not. We should assume that the env file will define the environment; i.e. if we are running in the dev environment the env file will contain dev values, if we are running in the prod environment, the env file will contain prod values etc.

Comment (davidfowl) [reply server/config/config.js:5]: That's what I'm thinking.

**Review by JChris246 — APPROVED** [review]
👍

**Pull request merged**